packages/@aws-cdk/aws-rds/lib/cluster.ts

```diff
@@ -542,23 +542,38 @@ export class DatabaseCluster extends DatabaseClusterNew {
    * Adds the single user rotation of the master password to this cluster.
    */
   public addRotationSingleUser(options: RotationSingleUserOptions = {}): secretsmanager.SecretRotation {
+    if (!options.secret || options.secret === this.secret) {
+      return this.addSingleUserRotationForMasterSecret(options);
+    }
+
+    const id = 'RotationSingleUser';
+    return this.addSingleUserRotationForSecret(options.secret as unknown as Construct, id, options.secret, options);
+  }
+
+  private addSingleUserRotationForMasterSecret(options: RotationSingleUserOptions): secretsmanager.SecretRotation {
     if (!this.secret) {
-      throw new Error('Cannot add single user rotation for a cluster without secret.');
+      throw new Error('Cannot add single user rotation for a cluster without master secret.');
     }
 
     const id = 'RotationSingleUser';
-    const existing = this.node.tryFindChild(id);
-    if (existing) {
-      throw new Error('A single user rotation was already added to this cluster.');
+    return this.addSingleUserRotationForSecret(this, id, this.secret, options);
+  }
+
+  private addSingleUserRotationForSecret(scope: Construct, id: string, secret: secretsmanager.ISecret, options: RotationSingleUserOptions)
+    : secretsmanager.SecretRotation {
+    // TODO: this is a hack! what's the right way to do it?
```

Comment:
Not sure what you mean... `Construct` definitely has the `.node` property...

Reply:
There's two `Construct`, one imported from `'constructs'`, one in `'construct-compat'` (extended by `Resource`). I'm using the former one that is imported into the file already and has no `node` field. I wasn't sure if I could/should use the other one. What's the correct way to do this?

Reply:
Ah, I see what you mean. For the `Construct` imported from `constructs`, the API is `constructs.Node.of(myConstruct)`.

Reply:
Done.